include/exec.h

```diff
@@ -10,6 +10,8 @@ typedef struct exec_args {
    * starting process. */
   unsigned argc;
   const char **argv;
+  unsigned envc;
```

Comment:
FYI There's no such thing as `envc`.

Reply:
Ok. But please keep in mind this is an optimization. Because execve does not provide `argc` or `envc` a kernel developer may choose wrong values for those. In my view this is violation of [SPOT](https://en.wikipedia.org/wiki/Single_source_of_truth) principle. I'd prefer for a procedure to calculate the values instead of relying on user to provide good ones.